docs: Add docs for Livecycle Docker extension (#294)
livecycle/preevy: site/docs/integrations/docker-extension.md

   ![Livecycle Docker Extension](docker-ext-5.png)

5. **Accessing Livecycle dashboard**

   You can also access the Livecycle dashboard to see the logs and debug your application. Click on the "Open Link" button to open the Livecycle dashboard. On the dashboard, you can see all the running applications and services. You can also see the public and private URLs for each service.

   ![Livecycle Docker Extension](docker-ext-6.png)

6. **Debugging, inspecting, and logging**

   Once you have opened the Livecycle dashboard, you can see all the environments/apps that are running. Click on the name of the environment for which you want to see the logs, terminal, etc. You can view the logs, terminal, and container inspection for each service.

   ![Livecycle Docker Extension](docker-ext-7.png)

That's it! You have successfully installed the Livecycle Docker Extension and shared your local development environment with your team.

## FAQ

<details>
  <summary>What is the difference between the Livecycle Docker Extension and ngrok?</summary>
  <p>

>     The main difference is the collaboration aspect. Livecycle allows you to share your local development environment with your team, so you can get feedback much earlier in the development workflow.

Author: ```suggestion
    The difference is that our extension serves as a gateway to demonstrate the collaboration experience you can achieve using Preevy in remote environments. If you find the collaboration flow useful, you can expand it to create preview environments using the same tool.
```

title: Livecycle Docker Extension
---

# Livecycle Docker Extension 🐳

## Overview

> Livecycle Docker Extension helps you to share your local development environment with your team. It allows you to share your local development environment with your team, so you can get feedback much earlier in the development workflow, without the hassle of staging environments or CI builds.

Reviewer: Repeated phrase + missing article

```suggestion
The Livecycle Docker Extension enables you to share your local development environment with your team, so you can get feedback much earlier in the development workflow, without the hassle of staging environments or CI builds.
```

## Key Features

- Instant, secure tunneling to frontend and backend services

> - Share HTTPS links so teammates can quickly access and review code while it's in flight

Reviewer: ```suggestion
- Share HTTPS links so teammates can quickly access and review your web UI and backend services at runtime.
```

> - Public and private access controls

Reviewer: ```suggestion
- Public and private access
```

> - Integration with external identity providers for secure access

Reviewer: I don't think we currently offer that feature

Reviewer 2: Maybe we need to rephrase it, the external identity is offered by Livecycle.run (with Google/Github and Livecycle Org)

Author: Hey, can you tell then what should say here?

- Integrated debugging tools that include log inspection, shell, and container inpsection

> - Integration with OSS Preevy CLI tool that can be used for publishing ephemerals environment on your cloud provider or any Kubernetes cluster

Reviewer: ```suggestion
- Integration with the [OSS Preevy CLI tool](https://github.com/livecycle/preevy) that can be used for publishing ephemeral environments on your cloud provider or any Kubernetes cluster
```

## Getting Started

### Prerequisites

- [Docker Desktop](https://www.docker.com/products/docker-desktop)

> The content is fairly well-written, but there are a few grammar and clarity improvements that can be made. Here's the revised content:

Reviewer: Remove

```suggestion
```

4. **Accessing the shared URL**

   Once you have shared the URL with your team members, they can access the URL in their browser. They will be able to see the application running in their browser.

   ![Livecycle Docker Extension](docker-ext-5.png)

5. **Accessing Livecycle dashboard**

   You can also access the Livecycle dashboard to see the logs and debug your application. Click on the "Open Link" button to open the Livecycle dashboard. On the dashboard, you can see all the running applications and services. You can also see the public and private URLs for each service.

   ![Livecycle Docker Extension](docker-ext-6.png)

6. **Debugging, inspecting, and logging**

   Once you have opened the Livecycle dashboard, you can see all the environments/apps that are running. Click on the name of the environment for which you want to see the logs, terminal, etc. You can view the logs, terminal, and container inspection for each service.

   ![Livecycle Docker Extension](docker-ext-7.png)

That's it! You have successfully installed the Livecycle Docker Extension and shared your local development environment with your team.

## FAQ

<details>

>   <summary>How Livecycle Docker Extension is different from ngrok?</summary>

Reviewer: ```suggestion
  <summary>What is the difference between the Livecycle Docker Extension and ngrok?</summary>
```

  <p>

>     The main difference is the collaboration aspect. Livecycle allows you to share your local development environment with your team, so you can get feedback much earlier in the development workflow.

Reviewer: I'm not sure that is the difference, as ngrok allows sharing your local dev env as well.

I think the difference is that our Extension is a gateway to demonstrate the collaboration experience you can get using Preevy on remote environments. if you find the collaboration flow useful, you can expand it to create preview environments using the same tool.

  </p>
</details>

<details>
  <summary>Which frameworks and languages does Livecycle support?</summary>
  <p>
    Livecycle is agnostic to specific language or framework. It works with any language or framework that can be run in a Docker container.
  </p>
</details>

> <details>
>   <summary>Is Livecycle Docker Extension secure?</summary>
>   <p>
>     Yes, Livecycle Docker Extension is secure. It uses a private tunnel to expose your local development environment to the internet. You can also add access controls to restrict access to your environment. In addition, Livecycle Docker Extension supports HTTPS and SSL termination.
>   </p>

Reviewer: This question can never be realistically answered with a bold "yes".

```suggestion
<details>
  <summary>What security features does the Livecycle Docker extension offer?</summary>
  <p>
    The Livecycle Docker Extension uses a secure [SSH tunnel](https://livecycle.io/blogs/preevy-proxy-service-2/) to expose your local development environment using Livecycle's tunnel server, which is only accessible using HTTPS. You can enable private URLs to restrict access to your environment.
  </p>
```

</details>

<details>

>   <summary>How do I get started with Livecycle Docker Extension?</summary>

Reviewer: ```suggestion
  <summary>How do I get started with the Livecycle Docker Extension?</summary>
```

  <p>

>     You can get started with Livecycle Docker Extension by following the steps in the <a href="#getting-started">Getting Started</a> section.

Reviewer: ```suggestion
    You can get started with the Livecycle Docker Extension by following the steps in the <a href="#getting-started">Getting Started</a> section.
```

  </p>
</details>

<details>

>   <summary>What makes Livecycle Docker Extension different from other tools?</summary>

Reviewer: ```suggestion
  <summary>What makes the Livecycle Docker Extension different from other tools?</summary>
```

  <p>

>     Support provisioning to your cloud provider or k8s cluster -> In that context we are not bound to abilities like tunnel or ngrok as we allow the environment to be available and accessible regardless of your local setup. We also support your CI pipeline- showing that it can be added to your GH actions for example while also add comments to your GH.

Reviewer: This explanation needs some clarification - you can't really do those things with just the extension.

```suggestion
    Support provisioning to your cloud provider or k8s cluster -> In that context we are not bound to abilities like tunnel or ngrok as we allow the environment to be available and accessible regardless of your local setup. We also support your CI pipeline- showing that it can be added to your GH actions for example while also add comments to your GH.
```

Author: Hey, thanks

Can you tell me what should I write instead of that?

  </p>
</details>

<details>

>   <summary>How do I get support for Livecycle Docker Extension?</summary>

Reviewer: ```suggestion
  <summary>How do I get support for the Livecycle Docker Extension?</summary>
```

  <p>

>     You can join the <a href="https://community.livecycle.io">Livecycle Community</a> on Slack to get support for Livecycle Docker Extension.

Reviewer: ```suggestion
    Join the <a href="https://community.livecycle.io">Livecycle Community</a> on Slack to get support for the Livecycle Docker Extension.
```

   ![Livecycle Docker Extension](docker-ext-2.png)

3. **Getting shareable URLs**
   
   As soon as you log in, you will be able to see a list of running docker compose applications and all the services that are running in them. To get a public shareable URL for every service, turn on the toggle next to the compose application name. After that, you will be prompted to choose the access level. 

   ![Livecycle Docker Extension](docker-ext-3.png)

   You can choose between public and private access. If you choose public access, you will get a public URL that you can share with anyone. If you choose private access, you will get a private URL that require authentication and can only by used by your organization mebers. Then click on the "Share" button to get the shareable URL.

   ![Livecycle Docker Extension](docker-ext-4.png)

4. **Accessing the shared URL**

   URLs created by the extension are consistent, shareable and can be use by a browser or any other http client.
   Using these URLs, your team members will be able to see and interact with your local version of the app as long as the tunnel is open your workstation is running.  

>    Private environments require adding team members to your organization, and upon access, your team members will be prompt to authenticate.

Reviewer: prompt -> prompted

   The Livecycle dashboard requires authentication and organization membership, similiarly to private environments/services 

   ![Livecycle Docker Extension](docker-ext-6.png)

6. **Debugging, inspecting, and logging**

   Once you have opened the Livecycle dashboard, you can see all the environments/apps that are running. Click on the name of the environment for which you want to see the logs, terminal, etc. You can view the logs, terminal, and container inspection for each service.

   ![Livecycle Docker Extension](docker-ext-7.png)

That's it! You have successfully installed the Livecycle Docker Extension and shared your local development environment with your team.

## FAQ

<details>
  <summary>What is the difference between the Livecycle Docker Extension and Ngrok?</summary>
  
  Livecycle Docker extesnion is integrated with Docker, and provides a smoother experience for Docker users.  
  Consistent URLs, private environments, organizations and Google/Github authentication are supported out of the box.  
  Livecycle dashboard provides debugging capabilites that includes log inspection, shell access, and container inspection.  

>   Lastly, the integration with Preevy allow you to create remote ephemeral environments that can be used when your workstation is offline or futher in your SDLC (preview environments for Pull Requests).  

Reviewer: IMO further in your SDLC is too jargon

Reviewer 2: I don't like it that much, as well, but it's the best I could come up with to indicate it can be used further in the development process.

</details>

<details>
  <summary>What security features does the Livecycle Docker extension offer?</summary>
  
  The Livecycle Docker Extension uses a secure [SSH tunnel](https://livecycle.io/blogs/preevy-proxy-service-2/) to expose your local development environment using Livecycle's tunnel server, which is only accessible using HTTPS.  
  You can enable private URLs to restrict access to your environment.
</details>

<details>
  <summary>How do I get started with the Livecycle Docker Extension?</summary>
  
  You can get started with the Livecycle Docker Extension by following the steps in the [Getting Started](#getting-started) section.
</details>

<details>
  <summary>What makes the Livecycle Docker Extension different from other tools?</summary>
  
  Livecycle Docker extension is integrated with Preevy CLI and Livecylce services.
  Using the CLI, you can provision remote ephemeral environments that can be used in addtion to sharing local environments.  
  This functionality can also be used in the CI to provision ephemeral environment for Pull Requests.  

>   Livecycle add additonal layers of management, collaboration, review and debugging on top of shared environments.  

Reviewer: add->adds

>   Livecycle add additonal layers of management, collaboration, review and debugging on top of shared environments.  

Reviewer: add->adds
